src/loader/mod.rs

```diff
@@ -288,20 +319,45 @@ pub struct BzImage;
 #[cfg(feature = "bzimage")]
 #[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
 impl KernelLoader for BzImage {
-    /// Loads a bzImage
+    /// Loads a kernel from a bzImage to guest memory.
     ///
-    /// kernel is loaded into guest memory at code32_start the default load address
-    /// stored in bzImage setup header.
+    /// The kernel is loaded at `code32_start`, the default load address stored in the bzImage
+    /// setup header.
     ///
     /// # Arguments
     ///
-    /// * `guest_mem` - The guest memory where the kernel image is loaded.
-    /// * `kernel_start` - The offset into 'guest_mem' at which to load the kernel.
+    /// * `guest_mem`: [`GuestMemory`] to load the kernel in.
+    /// * `kernel_start`: Address in guest memory where the kernel is loaded.
     /// * `kernel_image` - Input bzImage image.
-    /// * `highmem_start_address` - This is the start of the high memory, kernel should above it.
+    /// * `highmem_start_address`: Address where high memory starts.
+    ///
+    /// # Examples
     ///
-    /// # Returns
-    /// * KernelLoaderResult
+    /// ```rust
+    /// # extern crate vm_memory;
+    /// # use linux_loader::loader::*;
+    /// # use vm_memory::{Address, GuestAddress, GuestMemoryMmap};
+    /// # use std::io::Cursor;
+    /// let mem_size: usize = 0x1000000;
+    /// let himem_start = GuestAddress(0x0);
+    /// let kernel_addr = GuestAddress(0x200000);
+    /// let gm = GuestMemoryMmap::from_ranges(&[(GuestAddress(0x0), mem_size)]).unwrap();
+    /// let mut kernel_image = vec![];
+    /// kernel_image.extend_from_slice(include_bytes!("bzimage"));
+    /// let loader_result = Elf::load(
+    ///     &gm,
+    ///     Some(kernel_addr),
+    ///     &mut Cursor::new(&kernel_image),
+    ///     Some(himem_start),
+    /// ).unwrap();
+    /// assert_eq!(loader_result.kernel_load.raw_value(), 0x200000);
+    /// assert_eq!(loader_result.setup_header.unwrap().header, 0x53726448);
+    /// assert_eq!(loader_result.setup_header.unwrap().version. 0x20d);
+    /// assert_eq!(loader_result.setup_header.unwrap().loadflags, 1);
+    /// assert_eq!(loader_result.kernel_end, 0x60c320);
```

Comment:
Can we drop these asserts? It makes the code example longer than it needs to be. These should be tested with unit test instead (and probably they are).

Reply:
Yup